### PR TITLE
Remove TypeChecker::ClosuresWithUncomputedCaptures

### PR DESCRIPTION
**Pull request by slavapestov**
Only TC.definedFunctions remains.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/AST/Decl.cpp [hunk 1]
```diff
@@ -2892,10 +2892,7 @@ bool ValueDecl::isRecursiveValidation() const {
 Type ValueDecl::getInterfaceType() const {
   auto &ctx = getASTContext();
 
-  // N.B. This assertion exists to catch new broken callers. It can be removed
-  // with the LazyResolver when the time comes.
-  assert(ctx.getLegacyGlobalTypeChecker()
```
Comment (CodaFi) [inline lib/AST/Decl.cpp]: Do we have to take this out?  It's catching SourceKit regressions right now #28226.

File: lib/Sema/ConstraintSystem.cpp [hunk 2]
```diff
@@ -3474,3 +3474,37 @@ ConstraintSystem::isConversionEphemeral(ConversionRestrictionKind conversion,
...
+    assert(isInDefaultArgumentContext);
+    assert(newClosureType
+               ->withExtInfo(newClosureType->getExtInfo().withNoEscape(true))
+               ->isEqual(closureType));
+    result = new (Context) FunctionConversionExpr(closure, closureType);
+  }
+
+  cacheExprTypes(result);
+  return result;
+}
```
Comment (CodaFi) [inline lib/Sema/ConstraintSystem.cpp]: Nit: trailing newline please.

**Review by CodaFi — APPROVED** [review]

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**